### PR TITLE
Adds custom pandas accessor for fitting cosmology

### PR DESCRIPTION
**Pull request by djperrefort**

### CODE REVIEW

**Comment by wmwv** [comment]
When fitting for cosmology, we marginalize over H0 and M_B.

But because this is just a constant offset to all data points, we can do this analytically.  We don't have to explicitly include that in the grid.

File: tests/test_cosmology_accessor.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    """Tests for the chi-squared minimization"""
+
+    def setUp(self):
+        self.test_data = mock.create_mock_pipeline_outputs()
+
+        # Arguments to ensure the minimization converges and quickly
+        self.iminuit_kwargs = dict(
+            H0=const.betoule_H0,
+            Om0=const.betoule_omega_m,
+            limit_Om0=(0, 1),
```
Comment (wmwv) [inline tests/test_cosmology_accessor.py:46]: When running the test suite I get that `limit_Om0` is not one of the expected parameters.
```
[Serenity SN-PWV] python -m unittest tests/test_cosmology_accessor.py
...EE.......
======================================================================
ERROR: test_correct_number_of_samples (tests.test_cosmology_accessor.MCResampling)
Test the correct number of samples are drawn from the dataset
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/wmwv/Research/mwvgroup/SN-PWV/tests/test_cosmology_accessor.py", line 57, in test_correct_number_of_samples
    samples = self.test_data.snat_sim.minimize_mc(num_samples, frac=1, **self.iminuit_kwargs)
  File "/Users/wmwv/Research/mwvgroup/SN-PWV/snat_sim/fitting_pipeline.py", line 375, in minimize_mc
    samples = [self.data.sample(n=n, frac=frac).snat_sim.minimize(**kwargs) for _ in range(samples)]
  File "/Users/wmwv/Research/mwvgroup/SN-PWV/snat_sim/fitting_pipeline.py", line 375, in <listcomp>
    samples = [self.data.sample(n=n, frac=frac).snat_sim.minimize(**kwargs) for _ in range(samples)]
  File "/Users/wmwv/Research/mwvgroup/SN-PWV/snat_sim/fitting_pipeline.py", line 347, in minimize
    minimizer = Minuit(self.chisq, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/iminuit/_minuit.py", line 485, in __init__
    self._init_state = _make_init_state(self._pos2var, args, kwds)
  File "/usr/local/lib/python3.9/site-packages/iminuit/_minuit.py", line 1401, in _make_init_state
    raise RuntimeError(
RuntimeError: limit_Om0 is not one of the parameters [H0 Om0 abs_mag w0 alpha beta]

======================================================================
ERROR: test_statistic_returns_dict (tests.test_cosmology_accessor.MCResampling)
Test supplying a statistic argument returns a dict of the applies statistic
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/wmwv/Research/mwvgroup/SN-PWV/tests/test_cosmology_accessor.py", line 66, in test_statistic_returns_dict
    samples = self.test_data.snat_sim.minimize_mc(num_samples, frac=1, **self.iminuit_kwargs)
  File "/Users/wmwv/Research/mwvgroup/SN-PWV/snat_sim/fitting_pipeline.py", line 375, in minimize_mc
    samples = [self.data.sample(n=n, frac=frac).snat_sim.minimize(**kwargs) for _ in range(samples)]
  File "/Users/wmwv/Research/mwvgroup/SN-PWV/snat_sim/fitting_pipeline.py", line 375, in <listcomp>
    samples = [self.data.sample(n=n, frac=frac).snat_sim.minimize(**kwargs) for _ in range(samples)]
  File "/Users/wmwv/Research/mwvgroup/SN-PWV/snat_sim/fitting_pipeline.py", line 347, in minimize
    minimizer = Minuit(self.chisq, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/iminuit/_minuit.py", line 485, in __init__
    self._init_state = _make_init_state(self._pos2var, args, kwds)
  File "/usr/local/lib/python3.9/site-packages/iminuit/_minuit.py", line 1401, in _make_init_state
    raise RuntimeError(
RuntimeError: limit_Om0 is not one of the parameters [H0 Om0 abs_mag w0 alpha beta]

----------------------------------------------------------------------
Ran 12 tests in 0.154s

FAILED (errors=2)
```

**Review by wmwv — CHANGES_REQUESTED** [review]
Code looks fine on inspection.  Test doesn't succeed.

Restructure to use analytic marginalization for H0 and M_B.

File: tests/test_cosmology_accessor.py (hunk 1 above)
Comment (djperrefort) [reply tests/test_cosmology_accessor.py:46]: I think this has to do with `iminuit`'s recent bump to 2.0. Should be a quick fix.

**Pull request closed without merging**